[docs] Add note about termination period for shutdown (#7751)
apollographql/router: docs/source/routing/self-hosted/containerization/kubernetes/other-considerations.mdx

    * Lower the cache size if the cache reaches near 100% hit-rate but the cache size is still growing.
    * Increase the pod memory if the cache hit rate is low and the cache size is still growing.
    * Lower the cache size if the latency of query planning cache misses is acceptable and memory availability is limited.

## Shutting down gracefully

Apollo Router stops accepting new requests immediately after receiving a `SIGTERM` in Kubernetes. It waits for all active requests to complete before shutting down. There is no time limit controlling this process; it continues until all requests either finish or time out at the request level.

> Using other features of [Kubernetes lifecycle hooks](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/), you can configure the [health checks](/graphos/routing/self-hosted/health-checks) to make sure all pods are shutdown safely. It may also be wise to consider setting the shut down grace period to be slightly longer than any  configured [router timeouts](/graphos/routing/performance/traffic-shaping).

Reviewer: ```suggestion
You can configure health checks using [Kubernetes lifecycle hooks](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/) to ensure all pods shut down safely. You can also set the shutdown grace period to be slightly longer than any configured [router timeouts](/graphos/routing/performance/traffic-shaping).
```
Taking Apollo Librarian suggestions, but these new links look great, thanks Shane!
